### DISCUSSION TITLE
Is it valid to register a field inside a component that is a controlled component with Controller

### DISCUSSION THREAD

**Discussion by gumpenaajayvarma** [react-hook-form/react-hook-form#6080] [Q&A]
Hi,

I have developed a custom component which has a table of form elements in it. I wrapped it with Controller of the hook form in order to manage the values to the fields.

e.g:
```
<Controller
    name='formTable'
    render={({ onChange, value }) => {
         return <FormTable
             onChange={onChange}
             value={value}
         />
    }}
/>
```
As part of another requirement I have to update all the values of the form table based on a dropdown whose options are dynamic. Hence I modified above code to this:
```
<Controller
    name='formTable'
    render={({ onChange, value }) => {
         return <FormTable
             onChange={onChange}
             value={value}
         >
             <Rows
                onChange={onChange}
                formTableValue={value}
             />
         </FormTable>
    }}
/>
```

In the Rows component I am registering the dropdown using Controller. Which is working. But facing issues with the dirty fields of those dropdowns in the Rows.

Is it valid to register fields in a component(in this case in the "Rows" component) that is wrapped inside a Controller(with name "formTable") ? I am using Hook form v6

If it is valid, why in my case the field is not listed as dirty when I choose a value that was initialized?
I am initializing my form using reset() for async values

CodeSandbox:
https://codesandbox.io/s/hungry-hill-qmz7u?file=/src/App.tsx

**Comment by gumpenaajayvarma** [comment]
Hi @bluebill1049,
Can you help me with this please?

I have a dropdown based on which I need to make an api call. I am ensuring that the api call is not made unless the field is dirty as to not override the values already saved. But If I choose a previous selected value the field is not showing up in dirty.

https://codesandbox.io/s/hungry-hill-qmz7u?file=/src/App.tsx

Expectation:
Able to initialize the dropdown to a value using reset ()
Should get the dropdown field listed in dirtyFields if user changes to other option and chooses back the initialized value as based on this an api call has to be made.

**Comment by bluebill1049** [comment]
you can not register input within a controller, because `Controller` is basically doing the register inside for you.

> if user changes to other option and chooses back the initialized value as based on this an api call has to be made.

not sure what do you mean above.

**Reply by gumpenaajayvarma** [reply]
Use case is as such:
Initial value of the dropdown is 2
I select "option 1" whose value is 1 => Then the field is shown in the dirtyFields
I select "option 2" back whose value is 2 => Then the field is not shown in the dirtyFields.

I am controlling the API call based on if the field is dirty or not. The response I get from API is used to update the FormTable component values.

**Reply by gumpenaajayvarma** [reply]
The sandbox in this link also explains my problem:
https://www.gitmemory.com/issue/react-hook-form/react-hook-form/3562/736253523

**Reply by gumpenaajayvarma** [reply]
Hey @bluebill1049 
Did you get a chance to check regarding the dirtyFields query?

**Reply by bluebill1049** [reply]
check your CSB: https://codesandbox.io/s/7o2wrp86k6?file=/src/index.js it's broken, can you get that updated and followed with steps on how to reproduce the problem. try to keep the CSB simple if you can as well.

**Reply by gumpenaajayvarma** [reply]
https://codesandbox.io/s/hungry-hill-qmz7u?file=/src/App.tsx is the sandbox. You can refer to it.

**Reply by gumpenaajayvarma** [reply]
I am expecting that the onChange of the dropdown field, it should be listed in dirtyFields. So that I can make an API call even for a previous selected default value.

**Reply by bluebill1049** [reply]
that's because you have reset the form, so does it's defaultValues 

```
oFormAPI.reset({
  selectOpt: "2",
  selectOpt2: "2"
});
```

**Reply by bluebill1049** [reply]
if you just want to update value, use `setValue` instead.

**Reply by gumpenaajayvarma** [reply]
I am using reset to GET the form values from API and initialize them.
